internal/pkg/db/mongo/export.go

```diff
@@ -67,28 +89,59 @@ func (mc MongoClient) UpdateRegistration(reg export.Registration) error {
 // Get a registration by ID
 // UnexpectedError - problem getting in database
 // NotFound - no registration with the ID was found
-func (mc MongoClient) RegistrationById(id string) (export.Registration, error) {
-	if !bson.IsObjectIdHex(id) {
-		return export.Registration{}, db.ErrInvalidObjectId
+func (mc MongoClient) RegistrationById(id string) (contract.Registration, error) {
+	reg, err := mc.registrationById(id)
+	if err != nil {
+		return contract.Registration{}, err
+	}
+	return reg.ToContract(), nil
+}
+
+func (mc MongoClient) registrationById(id string) (models.Registration, error) {
+	var query bson.M
+	if !bson.IsObjectIdHex(id){
+		// Id is not a BSON ID, is it a UUID?
+		_, err := uuid.Parse(id)
+		if err != nil {
+			return models.Registration{}, db.ErrInvalidObjectId
+		}
+		query = bson.M{"uuid": id}
+	} else {
+		query = bson.M{"_id": bson.ObjectIdHex(id)}
 	}
-	return mc.getRegistration(bson.M{"_id": bson.ObjectIdHex(id)})
+
+	reg, err := mc.getRegistration(query)
+	if err != nil {
+		return models.Registration{}, err
+	}
+
+	return reg, nil
 }
 
 // Get a registration by name
 // UnexpectedError - problem getting in database
 // NotFound - no registration with the name was found
-func (mc MongoClient) RegistrationByName(name string) (export.Registration, error) {
-	return mc.getRegistration(bson.M{"name": name})
+func (mc MongoClient) RegistrationByName(name string) (contract.Registration, error) {
+	reg, err := mc.registrationByName(name)
```

Comment:
I would inline `registrationByName` here and refactor `RegistrationByName` as this is its only reference.